processing/src/main/java/io/druid/query/search/SearchQueryQueryToolChest.java

```diff
@@ -208,33 +225,82 @@ public Object apply(Result<SearchResultValue> input)
           public Result<SearchResultValue> apply(Object input)
           {
             List<Object> result = (List<Object>) input;
+            boolean needsRename = false;
+            final Map<String, String> outputNameMap = Maps.newHashMap();
+            if (dimensionSpecs != null) {
+              List<String> cachedOutputNames = (List) result.get(2);
```

Comment:
This could be null if we're pulling a cache value written by an old version of Druid to a shared cache (like memcached). One way we could handle this by adjusting the search query cache key in such a way that old cached values will never be pulled (add some incompatibility to the cache key).